lib/collection/src/shards/replica_set.rs

```diff
@@ -1755,6 +1755,23 @@ impl ShardReplicaSet {
         .await
     }
 
+    pub async fn custom_search(
```

Comment:
Would `core_search` make more sense?

```suggestion
    pub async fn core_search(
```